larray/tests/test_array.py

```diff
@@ -2648,10 +2648,31 @@ def test_to_excel_xlwings(self):
 
     @pytest.mark.skipif(xw is None, reason="xlwings is not available")
     def test_open_excel(self):
-        # 1) with headers
-        # ===============
+        # 1) Mode
+        # =======
+        # 1a) Read and read+ modes crash if file does not exist
+        with pytest.raises(ValueError):
+            open_excel(abspath('test_open_excel.xlsx'), mode='r')
```

Comment:
In these kind of case I prefer when the filename makes it obvious that it does not exist (it would also make sure we do not accidentally break the test with another test creating this file). eg. 'this_file_does_not_exist.xlsx'